third_party/2.7/dateutil/parser.pyi

```diff
@@ -0,0 +1,37 @@
+from typing import List, Tuple, Optional, Callable, Union, IO, Any
+from datetime import datetime
+
+__all__ = ...  # type: List[str]
+
+
+class parserinfo(object):
+    JUMP = ... # type: List[str]
+    WEEKDAYS = ... # type: List[Tuple[str, str]]
+    MONTHS = ... # type: List[Tuple[str, str]]
+    HMS = ... # type: List[Tuple[str, str, str]]
+    AMPM = ... # type: List[Tuple[str, str]]
+    UTCZONE = ... # type: List[str]
+    PERTAIN = ... # type: List[str]
+    TZOFFSET = ... # type: Dict[str, int]
+
+    def __init__(self, dayfirst: bool=..., yearfirst: bool=...) -> None: ...
+    def jump(self, name: str) -> bool: ...
+    def weekday(self, name: str) -> Union[int, None]: ...
+    def month(self, name: str) -> Union[int, None]: ...
+    def hms(self, name: str) -> Union[int, None]: ...
+    def ampm(self, name: str) -> Union[int, None]: ...
+    def pertain(self, name: str) -> bool: ...
+    def utczone(self, name: str) -> bool: ...
+    def tzoffset(self, name: str) -> Union[int, None]: ...
+    def convertyear(self, year: int) -> int: ...
+    def validate(self, res: datetime) -> bool: ...
+
+class parser(object):
+    def __init__(self, info: Optional['parserinfo']) -> None: ...
+    def parse(self, timestr: str, default: Optional[datetime],
```

Comment:
On Mon, Jul 4, 2016 at 1:45 PM, Paul Ganssle <notifications@github.com>
wrote:

> I'm not super familiar with how the type annotations work (which is one
> reason I haven't done these annotations myself yet), but in the Python 3
> annotations, I think IO was used (see here
> <https://github.com/python/typeshed/blob/master/third_party/3/dateutil/parser.pyi#L35>).
> Looks like maybe IO[AnyStr] is better than IO[any], if I understand this
> notation correctly.
>

Thanks. I get a bit befuddled when considering differences between 2.x and
3.x, since I don't actually use 3.x. If I read that right, I don't think
the bytes type would be appropriate for a 2.x interface.

Does it matter that IO[AnyStr] has a whole bunch of other methods which
aren't read()? Is there some way to tell the type system that only the
read() method is important in parser.parse?

Skip